### DISCUSSION TITLE
[Kavita] Configurable Collapse of Series UI Metadata Elements

### DISCUSSION THREAD

**Discussion by Citizen7** [Kareadita/Kavita#2614] [Ideas]
## Current Situation
In the current UI, when opening a series, the Genre, Collections, Reading Lists, Writers, Cover Artists, Characters, Colorists, Editors, Inkers, Letterers, Pencillers, and Publisher metadata tag fields are all displayed by default before the actual issues/volumes section. Although each section displays only 30 tags before showing an [and X more] tag, these sections can easily grow to extremely large size, particularly when viewing long-running series.

![Screen capture of user scrolling through tags for the Superman comic series](https://github.com/Kareadita/Kavita/assets/5089643/86b3c906-615b-4976-a090-2bf5aa6cc5ca)

While there is a toggle at the bottom allowing the user to "See Less", this toggle does not persist when reloading the series, nor can it be set in the UI as the default.

## Proposed Solution
* Each section can be individually collapsed
* The default state (open, collapsed) can be set in the UI Customization screen

#### Potential Additional Features
*  Default state could be overwritten with per series settings
*  Current state remembered per series, instead of resetting to default on every load
*  Customize how many tags appear before [and X more]
*  The ability to re-order the metadata sections.

## Why This Is Important
Different users view metadata with different priorities. Some feel that it is of highest importance, while others find it significantly less helpful when it physically imposes itself between the reader and the material to be read. Being able to customize how much metadata is shown by default would allow users to prioritize for themselves to get the best reading experience.

### Duration of Using Kavita

6 months

### Idea Category

Feature Enhancement

### Agreement

- [X] I agree that this is solely for submitting ideas, and I will search for existing ideas before posting.

**Comment by majora2007** [comment]
Actually, this type of expand by default feature just shipped and on the changelog I asked for feedback. When originally implementing, users wanted more tags expanded. 

I'm not sure about building that level of customization for this (let's see community engagement), but what about adjusting the tag amount before automatically expanding the section?

**Reply by Citizen7** [reply]
That could certainly make some series more navigable.

**Reply by Citizen7** [reply]
I apologize, what version did that feature ship in? I haven't been able to find it by scrolling through the release logs.

**Reply by majora2007** [reply]
> Changed: On Desktop, by default all series metadata will show (for series metadata). I'm looking for feedback from users.
from https://github.com/Kareadita/Kavita/releases/tag/v0.7.11

**Reply by Citizen7** [reply]
Thank you. If there is a particular place you would like feedback to be collected for that (I didn't see one linked), I will put my opinion there. Otherwise, my feedback is that I would prefer them to be collapsed by default instead of shown, but I can see why other users would prefer the opposite, so I still think the ability to specify that per system (if not per user, or even per series) would be a nice feature.

**Reply by majora2007** [reply]
Here is completely fine. Thanks for sharing your feedback. I hope more people comment so we can find a good out of the box experience for everyone.

**Reply by lyingloverr** [reply]
I would prefer them to be collapsed by default as well.

**Reply by majora2007** [reply]
For future people, please upvote lyingloverr's preference if you're in agreement.
